components/ui/VersionFilterControl.vue

```diff
@@ -131,8 +154,37 @@ export default {
               projectVersion.loaders.includes(loader)
             ))
       )
+      this.updateQuery()
       this.$emit('updateVersions', temp)
     },
+    validateFilters() {
+      this.selectedLoaders = this.selectedLoaders.filter((loader) =>
+        this.getValidLoaders().includes(loader)
+      )
+      this.selectedGameVersions = this.selectedGameVersions.filter((version) =>
+        this.getValidVersions().some(
+          (validVersion) => validVersion.version === version
+        )
+      )
+    },
+    updateQuery() {
+      this.$router
+        .replace({
+          query: {
+            ...this.$route.query,
+            loaders: this.getAsQuery(this.selectedLoaders),
+            gameVersions: this.getAsQuery(this.selectedGameVersions),
+            showSnapshots: this.showSnapshots ? true : undefined,
+          },
+        })
+        .catch(() => {})
+    },
+    getAsQuery(elements) {
+      if (elements.length === 0) {
+        return undefined
+      }
+      return elements.join(',')
+    },
```

Comment:
just for code quality I think this func is fine being inlined with a ternary in both places